Code/Framework/AzToolsFramework/AzToolsFramework/ComponentMode/ComponentModeDelegate.cpp

```diff
@@ -229,13 +230,23 @@ namespace AzToolsFramework
             EntitySelectionEvents::Bus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
             EditorEntityVisibilityNotificationBus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
             EditorEntityLockComponentNotificationBus::Handler::BusConnect(entityComponentIdPair.GetEntityId());
+
+            AzFramework::ComponentModeDelegateNotificationBus::Broadcast(
+                &AzFramework::ComponentModeDelegateNotificationBus::Events::OnComponentModeDelegateConnect, m_entityComponentIdPair);
         }
 
         void ComponentModeDelegate::Disconnect()
         {
+            m_componentType = AZ::Uuid::CreateNull();
+            m_entityComponentIdPair = AZ::EntityComponentIdPair(AZ::EntityId(), AZ::InvalidComponentId);
+            m_handler = nullptr;
+
             EditorEntityLockComponentNotificationBus::Handler::BusDisconnect();
             EditorEntityVisibilityNotificationBus::Handler::BusDisconnect();
             EntitySelectionEvents::Bus::Handler::BusDisconnect();
+
+            AzFramework::ComponentModeDelegateNotificationBus::Broadcast(
+                &AzFramework::ComponentModeDelegateNotificationBus::Events::OnComponentModeDelegateDisconnect, m_entityComponentIdPair);
```

Comment:
Oh no sorry this won't work, we need to call this before we set `m_entityComponentIdPair` to invalid